### PR TITLE
Replace FLOC with browsingTopics

### PR DESCRIPTION
**Pull request by SlayterDev**
https://app.asana.com/0/1163321984198618/1201719300245228/f

`browsingTopics` is effectively replacing `floc`. This PR turns our `floc` protection into a `browsingTopics` protection.

New to this build process. Hopefully this looks right.

### CODE REVIEW

File: src/features/google-rejected.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+export function init () {
+    if ('browsingTopics' in Document.prototype) {
```
Comment (jonathanKingston) [inline src/features/google-rejected.js:2]: If we make the if statement inside the try, we don't need to do 3 of them aka:

```
try {
    if ('browsingTopics' in Document.prototype) {
        ....
    }
    if (...) {
        ...
    }
} catch {
    // Throw away this exception, it's likely a confict with another extension
}
```

File: src/features/google-rejected.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        try {
+            delete Document.prototype.browsingTopics
+        } catch {
+            // Throw away this exception, it's likely a confict with another extension
+        }
+    }
+
+    if ('joinAdInterestGroup' in Navigator.prototype) {
+        try {
+            delete Navigator.prototype.joinAdInterestGroup
```
Comment (jonathanKingston) [inline src/features/google-rejected.js:12]: I think there's some more here https://source.chromium.org/chromium/chromium/src/+/main:third_party/blink/renderer/modules/ad_auction/navigator_auction.idl:

```
  [CallWith=ScriptState, Measure, RaisesException]
  void joinAdInterestGroup(AuctionAdInterestGroup group, double durationSeconds);

  [CallWith=ScriptState, Measure, RaisesException]
  void leaveAdInterestGroup(optional AuctionAdInterestGroup group);

  [CallWith=ScriptState, Measure, RaisesException]
  void updateAdInterestGroups();

  [RuntimeEnabled=Fledge, CallWith=ScriptState, Measure, RaisesException]
  Promise<USVString?> runAdAuction(AuctionAdConfig config);

  [RuntimeEnabled=Fledge, CallWith=ScriptState, Measure, RaisesException]
  sequence<USVString> adAuctionComponents([Clamp] unsigned short numComponents);

```

**Review by jonathanKingston — CHANGES_REQUESTED** [review]
I think we're missing a few APIs here.

**Review by jonathanKingston — APPROVED** [review]

**Review by Pitpul1 — COMMENTED** [review]
Wel whel

**Pull request merged**